src/artm/score/perplexity.cc

```diff
@@ -33,52 +33,66 @@ void Perplexity::AppendScore(
     Score* score) {
   int topic_size = p_wt.topic_size();
 
-  // the following code counts perplexity
-  bool use_classes_from_model = false;
-  if (config_.class_id_size() == 0) use_classes_from_model = true;
-
-  std::map< ::artm::core::ClassId, float> class_weights;
-  if (use_classes_from_model) {
-    for (int i = 0; (i < args.class_id_size()) && (i < args.class_weight_size()); ++i)
-      class_weights.insert(std::make_pair(args.class_id(i), args.class_weight(i)));
+  // fields od proto mesage for all cases
+  std::unordered_map<::artm::core::ClassId, float> class_weight_map;
+  std::unordered_map<::artm::core::ClassId, double> normalizer_map;
+  std::unordered_map<::artm::core::ClassId, double> raw_map;
+  std::unordered_map<::artm::core::ClassId, ::google::protobuf::int64> zero_words_map;
+
+  double normalizer = 0.0;
+  double raw = 0.0;
+  ::google::protobuf::int64 zero_words = 0;
+
+  // choose class_ids policy
+  if (config_.class_id_size() == 0) {
+    for (int i = 0; (i < args.class_id_size()) && (i < args.class_weight_size()); ++i) {
+      class_weight_map.insert(std::make_pair(args.class_id(i), args.class_weight(i)));
+      normalizer_map.insert(std::make_pair(args.class_id(i), 0.0));
+      raw_map.insert(std::make_pair(args.class_id(i), 0.0));
+      zero_words_map.insert(std::make_pair(args.class_id(i), 0));
+    }
   } else {
     for (auto& class_id : config_.class_id()) {
-      for (int i = 0; (i < args.class_id_size()) && (i < args.class_weight_size()); ++i)
+      for (int i = 0; (i < args.class_id_size()) && (i < args.class_weight_size()); ++i) {
         if (class_id == args.class_id(i)) {
-          class_weights.insert(std::make_pair(args.class_id(i), args.class_weight(i)));
+          class_weight_map.insert(std::make_pair(args.class_id(i), args.class_weight(i)));
+          normalizer_map.insert(std::make_pair(args.class_id(i), 0.0));
+          raw_map.insert(std::make_pair(args.class_id(i), 0.0));
+          zero_words_map.insert(std::make_pair(args.class_id(i), 0));
           break;
         }
+      }
+    }
+    if (class_weight_map.empty()) {
+      LOG(ERROR) << "All requested class ids are not presented in model. Score calculation will be skipped";
```

Comment:
@MelLain Is it better to write `None of requested classes were presented in model. ...`?